patches/server/0819-Fix-kelp-modifier-changing-growth-for-other-crops.patch

```diff
@@ -0,0 +1,104 @@
++            modifier = world.spigotConfig.twistingVinesModifier;
++        } else if (state.is(Blocks.WEEPING_VINES) || state.is(Blocks.WEEPING_VINES_PLANT)) {
++            modifier = world.spigotConfig.weepingVinesModifier;
++        } else if (state.is(Blocks.CAVE_VINES) || state.is(Blocks.CAVE_VINES_PLANT)) {
++            modifier = world.spigotConfig.caveVinesModifier;
++        } else if (state.is(Blocks.KELP) || state.is(Blocks.KELP_PLANT)) {
++            modifier = world.spigotConfig.kelpModifier;
++        } else {
++            modifier = 100; // Above cases are exhaustive as of 1.18
++        }
++        if ((Integer) state.getValue(GrowingPlantHeadBlock.AGE) < 25 && random.nextDouble() < (100.0D / modifier) * this.growPerTickProbability) { // Spigot
++            // Paper end
+             BlockPos blockposition1 = pos.relative(this.growthDirection);
+ 
+             if (this.canGrowInto(world.getBlockState(blockposition1))) {
+-                org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockSpreadEvent(world, pos, blockposition1, this.getGrowIntoState(state, world.random)); // CraftBukkit
++                org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockSpreadEvent(world, pos, blockposition1, this.getGrowIntoState(state, world.random, world)); // CraftBukkit // Paper
+             }
+         }
+ 
+     }
+ 
++    // Paper start
++    protected BlockState getGrowIntoState(BlockState state, Random random, Level level) {
++        return this.getGrowIntoState(state, random);
++    }
++    // Paper end
++
+     protected BlockState getGrowIntoState(BlockState state, Random random) {
+         return (BlockState) state.cycle(GrowingPlantHeadBlock.AGE);
+     }
+diff --git a/src/main/java/org/spigotmc/SpigotWorldConfig.java b/src/main/java/org/spigotmc/SpigotWorldConfig.java
+index 9f7541cb62600f022da75cba74731ff4e57f7f36..f144ca888518f1bdb84ee811410937cba994245c 100644
+--- a/src/main/java/org/spigotmc/SpigotWorldConfig.java
++++ b/src/main/java/org/spigotmc/SpigotWorldConfig.java
+@@ -103,6 +103,12 @@ public class SpigotWorldConfig
+     public int bambooModifier;
+     public int sweetBerryModifier;
+     public int kelpModifier;
++    // Paper start
++    public int twistingVinesModifier;
++    public int weepingVinesModifier;
++    public int caveVinesModifier;
++    public int glowBerryModifier;
++    // Paper end
+     private int getAndValidateGrowth(String crop)
+     {
+         int modifier = this.getInt( "growth." + crop.toLowerCase(java.util.Locale.ENGLISH) + "-modifier", 100 );
+@@ -133,6 +139,12 @@ public class SpigotWorldConfig
+         this.bambooModifier = this.getAndValidateGrowth( "Bamboo" );
+         this.sweetBerryModifier = this.getAndValidateGrowth( "SweetBerry" );
+         this.kelpModifier = this.getAndValidateGrowth( "Kelp" );
++        // Paper start
++        this.twistingVinesModifier = this.getAndValidateGrowth("TwistingVines");
```

Comment:
Gotta put spaces between the parens and the string :stuck_out_tongue:

Reply:
paperweight will accept PRs if anyone can write a Mercury visitor that undoes that

Reply:
![image](https://user-images.githubusercontent.com/1228900/144692072-adf2645b-d6c6-4c49-92a2-e266f84e067b.png)